packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts

```diff
@@ -92,50 +91,72 @@ class DefaultMutationMapperStore implements MutationMapperStore {
     @observable
     private _activeTranscript: string | undefined = undefined;
 
+    @observable
+    private _selectedTranscript: string | undefined = undefined;
+
     constructor(
         public gene: Gene,
         protected config: DefaultMutationMapperStoreConfig,
         protected getMutations: () => Mutation[]
     ) {}
 
-    @computed
-    public get activeTranscript(): string | undefined {
-        let activeTranscript;
-        const canonicalTranscriptId =
-            !this.canonicalTranscript.isPending &&
-            this.canonicalTranscript.result
-                ? this.canonicalTranscript.result.transcriptId
-                : undefined;
-
-        if (this._activeTranscript !== undefined) {
-            activeTranscript = this._activeTranscript;
-        } else if (this.config.filterMutationsBySelectedTranscript) {
-            if (
-                this.transcriptsWithAnnotations.result &&
-                this.transcriptsWithAnnotations.result.length > 0 &&
-                canonicalTranscriptId &&
-                !this.transcriptsWithAnnotations.result.includes(
-                    canonicalTranscriptId
-                )
-            ) {
-                // if there are annotated transcripts and activeTranscript does
-                // not have any, change the active transcript
-                activeTranscript = this.transcriptsWithAnnotations.result[0];
-            } else {
-                activeTranscript = canonicalTranscriptId;
-            }
-        } else {
-            activeTranscript = canonicalTranscriptId;
-        }
+    readonly activeTranscript: MobxPromise<string | undefined> = remoteData(
+        {
+            await: () => [this.allTranscripts, this.canonicalTranscript],
+            invoke: async () => {
+                let activeTranscript;
+                const canonicalTranscriptId = this.canonicalTranscript.result
+                    ? this.canonicalTranscript.result.transcriptId
+                    : undefined;
 
-        // TODO this.activeTranscript = activeTranscript?
-        return activeTranscript;
-    }
+                const selectedTranscript =
+                    this._selectedTranscript &&
+                    this.allTranscripts.result &&
+                    this.allTranscripts.result
+                        .map(transcript => transcript.transcriptId)
```

Comment:
rather than doing map and then includes, look at _.some method (lodash).    it's a little more readable and performance is better (although here it's totally negligible).